src/main/java/io/vertx/core/http/impl/HttpClientImpl.java

```diff
@@ -1030,6 +1030,17 @@ private HttpClientRequest createRequest(HttpMethod method, String protocol, Stri
     return req;
   }
 
+  private boolean useProxy(boolean isSsl, ProxyOptions proxyOptions, String targetHost) {
```

Comment:
This functionality is duplicated in `HttpChannelConnector.doConnect()` that you also updated. Please move this helper method to e.g. a new class `src/main/java/io/vertx/core/http/impl/ProxyHelper.java`. Then reuse that helper in `doConnect()` as well. (You'll have to extract the proxy type frokm the proxyOptions instead, but that's fine.)